src/Gui/CommandFeat.cpp

```diff
@@ -136,33 +136,47 @@ StdCmdSendToPythonConsole::StdCmdSendToPythonConsole()
 
 bool StdCmdSendToPythonConsole::isActive(void)
 {
-    return (Gui::Selection().size() == 1);
+    //active only if either 1 object is selected or multiple subobjects from the same object
+    int size = Gui::Selection().size();
+    int xsize = size > 0 ? Gui::Selection().getSelectionEx()[0].getSubNames().size() : 0;
+    return (size == 1 || (size >= 1 && size == xsize));
 }
 
 void StdCmdSendToPythonConsole::activated(int iMsg)
 {
     Q_UNUSED(iMsg);
-
-    const std::vector<Gui::SelectionObject> &sels = Gui::Selection().getSelectionEx("*",App::DocumentObject::getClassTypeId(),true,true);
+    const std::vector<Gui::SelectionObject> &sels = Gui::Selection().getSelectionEx("*",App::DocumentObject::getClassTypeId(),true,false);
     if (sels.empty())
         return;
     const App::DocumentObject *obj = sels[0].getObject();
     QString docname = QString::fromLatin1(obj->getDocument()->getName());
     QString objname = QString::fromLatin1(obj->getNameInDocument());
     try {
-        QString cmd = QString::fromLatin1("obj = App.getDocument(\"%1\").getObject(\"%2\")").arg(docname,objname);
+        // clear variables from previous run, if any
+        QString cmd = QLatin1String("try:\n    del(doc,obj,shp,sub,subs)\nexcept:\n    pass\n");
+        Gui::Command::runCommand(Gui::Command::Gui,cmd.toLatin1());
+        cmd = QString::fromLatin1("doc = App.getDocument(\"%1\")").arg(docname);
+        Gui::Command::runCommand(Gui::Command::Gui,cmd.toLatin1());
+        cmd = QString::fromLatin1("obj = doc.getObject(\"%1\")").arg(objname);
         Gui::Command::runCommand(Gui::Command::Gui,cmd.toLatin1());
         if (sels[0].hasSubNames()) {
             std::vector<std::string> subnames = sels[0].getSubNames();
             if (obj->getPropertyByName("Shape")) {

```

Comment:
It should be avoided to use an explicit property type (here Shape) for geometric objects. Therefore the class App::GeoFeature offers the method getPropertyOfGeometry() that returns the geometric property.

The current implementation has two weaknesses:
* all other feature types except of Part features are not supported
* a Python feature could be implemented that has a property called Shape but in fact e.g. represents an integer

So, the suggested changes you have to make:
* for the given App::DocumentObject check if it's derived from App::GeoFeature and do a cast
* call getPropertyOfGeometry() and make sure that it's not null
* you get property name with getName() of the above result

Reply:
Okay.  I have made this change.  With this we can also get the mesh object's equivalent of the Shape property, too.  I had not thought of meshes.

Reply:
Currently the Mesh data structure hasn't an as advanced sub-element handling as Shapes but this could change in the future. The same applies to FE meshes or point clouds.